expression/expression.go

```diff
@@ -227,15 +266,17 @@ func VecEvalBool(ctx sessionctx.Context, exprList CNFExprs, input *chunk.Chunk,
 		nulls = append(nulls, false)
 	}
 
-	sel := selPool.Get().([]int)
-	defer selPool.Put(sel)
+	sel := allocSelSlice(n)
+	defer deallocateSelSlice(sel)
 	sel = sel[:0]
 	for i := 0; i < n; i++ {
 		sel = append(sel, i)
 	}
-	defer input.SetSel(input.Sel())
 	input.SetSel(sel)
 
+	// In areZeros slice, -1 means Null, 0 means zero, 1 means not zero
+	areZeros := allocZeroSlice(n)
```

Comment:
It's better to `s/areZero/isZero/`, so the code `if isZero[i]` would be more natural.